sos/report/plugins/ceph_podman.py

```diff
@@ -0,0 +1,110 @@
+        ])
+
+        if self._container_mon_name:
+            cmds = [
+                'ceph mon stat',
+                'ceph mon_status',
+                'ceph quorum_status',
+                'ceph mgr module ls',
+                'ceph mgr metadata',
+                'ceph osd metadata',
+                'ceph osd erasure-code-profile ls',
+                'ceph report',
+                'ceph osd crush show-tunables',
+                'ceph versions',
+                'ceph features',
+                'ceph osd crush dump',
+                'ceph -v',
+                'ceph crash stat',
+                'ceph crash ls',
+                'ceph config log',
+                'ceph config generate-minimal-conf',
+                'ceph config-key dump',
+                'ceph status',
+                'ceph health detail',
+                'ceph osd tree',
+                "ceph osd stat",
+                "ceph osd df tree",
+                "ceph osd dump",
+                "ceph osd df",
+                "ceph osd perf",
+                "ceph osd blocked-by",
+                "ceph osd pool ls detail",
+                "ceph osd numa-status",
+                "ceph device ls",
+                "ceph mon dump",
+                "ceph mgr dump",
+                "ceph mds stat",
+                "ceph df",
+                "ceph df detail",
+                "ceph fs ls",
+                "ceph fs dump",
+                "ceph pg dump",
+                "ceph pg stat",
+            ]
+            container_cmds = self.to_container_cmd(cmds,
+                                                   self._container_mon_name)
+            self.add_cmd_output(container_cmds, foreground=True)
+
+        if self._container_osd_name:
+            cmds = [
+                'ceph -v',
+                'ceph-volume lvm list',
+            ]
+            osd_containers = self.get_all_containers_by_name('ceph-osd-*')
+            for c in osd_containers:
+                container_cmds = self.to_container_cmd(cmds, c)
+                self.add_cmd_output(container_cmds, foreground=True)
+
+        self.add_forbidden_path([
+            "/etc/ceph/*keyring*",
+            "/var/lib/ceph/*keyring*",
+            "/var/lib/ceph/*/*keyring*",
+            "/var/lib/ceph/*/*/*keyring*",
+            "/var/lib/ceph/osd",
+            "/var/lib/ceph/mon",
+            "/var/lib/ceph/tmp/*mnt*",
+            "/etc/ceph/*bindpass*"
+        ])
+
+    def to_container_cmd(self, cmds, container_name):
+        container_cmds = []
+        for cmd in cmds:
+            container_cmds.append('podman exec %s %s' % (container_name, cmd))
+        return container_cmds
+
+    def get_all_containers_by_name(self, name):
+        all_containers = self.get_containers(get_all=True)
+        containers_by_name = []
+        for c in all_containers:
+            if re.match(name, c[1]):
+                containers_by_name.append(c[1])
+        return containers_by_name
```

Comment:
This can all be condensed down to

```python
return [c for c in self.get_containers() if re.match(name, c[1])]
```

Also, be aware that using `get_all=True` will return terminated containers as well, which you won't be able to run commands inside of.

Reply:
Thank you the suggestion, and pointing out `get_all=True`, will modify this part.